src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -107,18 +114,27 @@ def _extract_messages(self) -> None:
                     exc_info=True,
                 )
                 continue
-
             try:
                 if self.__input_codec:
                     self.__input_codec.validate(parsed_payload)
             except ValidationError:
                 # For now while this is still experimental, those errors are
                 # not supposed to be fatal.
-                logger.warn(
+                logger.warning(
                     "process_messages.invalid_schema",
                     extra={"payload_value": str(msg.payload.value)},
                     exc_info=True,
                 )
+            if use_case_id := extract_use_case_id(parsed_payload["name"]):
```

Comment:
So does the use case ID come from some property in the message now? If that's the case how come the use case ID is also passed to the constructor of IndexerBatch.

Reply:
For generic metrics, the use case IDs for haves been defined as the `Namespace` inside the name of the metric (more info on it [here](https://getsentry.github.io/relay/relay_metrics/struct.Metric.html#structfield.name)), but for `release_health` and `performance`... which is all we're getting right now there's some a 1:1 mapping.
| Namespace      | Use case ID |
| ----------- | ----------- |
| "PERFORMANCE" | something else |
| "RELEASE_HEALTH" | something else |
| everything else | the same |

But that pr is not merged in yet.

I updated TODO section in description!

Reply:
I think Lyn's right, since we're now extracting the `use_case_id` directly from the message payload, we won't need to pass it into the constructor anymore. See also my comment below about removing the hardcoded addition in `reconstruct_messages`.

With regards to the Namespace -> UseCaseID mapping:
This PR can use the `UseCaseID` class/enum for validation once that functionality is merged. 
For Performance, the Namespace in the metric name (MRI) is "transactions", and for Release Health that Namespace is "sessions."